### ISSUE TITLE
Handling Rest Request in Separte Thread

### ISSUE THREAD

**Issue by mersen** [oktal/pistache#253]
How to handle rest request in separate thread properly. I prepared exemplary code as below:

#include <pistache/endpoint.h>
using namespace Pistache;
struct HelloHandler : public Http::Handler
{
    HTTP_PROTOTYPE(HelloHandler)
    void onRequest(const Http::Request& request, Http::ResponseWriter writer) {

    std::cout << " Handle Request" << std::endl;
    std::thread thr([](Pistache::Http::ResponseWriter resW)
        {
            std::cout << "In thread" << std::endl;
            resW.send(Pistache::Http::Code::Ok, "{\"test\" : \"test1\"}");
        }, std::move(writer));
        thr.detach();
    }
};

int main() {
    Http::listenAndServe<HelloHandler>("*:9080");
}

 Pistache server responds properly and then terminates itself with:

terminate called after throwing an instance of 'Pistache::Async::BadType'
  what():  Argument type can not be used to resolve the promise  (TypeId does not match)
Aborted (core dumped)

I checked also std::move(writer.clone()) instead of std::move(writer) and nothing changed.

Does anyone know what is wrong?

**Comment by hoang408** [comment]
mersen, did you get pistache to respond to request from other computer?  I am trying to get it to work to no avail.

**Comment by dennisjenkins75** [comment]
hoang408: Pistache.io works fine over a network connection from other computers.

Have you verified that it is listening on the expected TCP post ("netstat -tanp" if on Linux)?

Have you verified the host firewall settings on both the client and server systems?

Do you have a network firewall between them?

Have you ran "tcpdump", "wireshark" or similar network sniffing tools on both computers while attempting your experiment?

Have you ran "netcat" in listen mode on the server computer, on the port that you wish to assign to pistache.io, and tried to connect to see if netcat (/usr/bin/nc) can accept a TCP connection on that port)?

Have you ran "strace", "ptrace" or "truss" or whatever task-level user/kernel syscall logger against your pistache.io process to observe its behavior while starting up and when receiving a TCP SYN?

**Comment by dennisjenkins75** [comment]
It seems that I commented in the wrong bug.  I apologize.  I'll move my comment.

**Comment by mersen** [comment]
I was not trying to do request to different computer

**Comment by mersen** [comment]
Problem was in separate commit taken to master branch form external repo. Problem is not present in master branch.
